File: data-structures/Python/queue.py
class Queue:
  def __init__(self):
    self.store = []

  def enqueue(self, element):
    self.store.append(element)
    return self

  def dequeue(self):
    # pop(0) removes the first element
    return self.store.pop(0)

  def is_empty(self):
    return true if len(self.store) == 0 else false

  def size(self):
    return len(self.store)

  def top(self):
    return self.store[0]

test = Queue()
test.enqueue("First")
test.enqueue("Second")
test.enqueue("Third")
test.dequeue()
test.dequeue()
test.enqueue("Stuff")
test.enqueue("Things")

print(test.store)
print(test.size())
print(test.top())

File: data-structures/JavaScript/queue.js
class Queue {
  constructor() {
    this.store = [];
  }

  // add an item to the end of the queue
  enqueue(data) {
    this.store.push(data);
    return this.store;
  }

  // remove an item from front of queue
  dequeue() {
    return this.store.shift();
  }

  // returns true if no items in queue
  isEmpty() {
    return this.store.length === 0 ? true : false;
  }

  // returns the number of items in the queue
  size() {
    return this.store.length;
  }

  // returns first element in the queue
  top() {
    return this.store[0];
  }
}

let test = new Queue;
test.enqueue(7);
test.enqueue(3);
test.enqueue(11);
console.log(test.store);
test.dequeue();
console.log(test.store);
console.log(test.size());
console.log(test.top());